### ISSUE TITLE
Update Python versions in the develop branch

### ISSUE THREAD

**Issue by Elliria** [autokey/autokey#773]
### AutoKey is a Xorg application and will not function in a Wayland session. Do you use Xorg (X11) or Wayland?

Xorg

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

Enhancement

### Which AutoKey GUI did you use?

None

### Can you briefly describe the issue?

All references to Python versions in the **develop** branch of the source code need to be updated to currently-supported versions.

### Can the issue be reproduced?

Always

### What are the steps to reproduce the issue?

Submit a pull request to the project.

### What should have happened?

It should pass all tests.

### What actually happened?

It fails on Python versions.

**Comment by josephj11** [comment]
Research:
    (All line numbers taken from 0.96.0 Beta 10)

See https://github.com/autokey/autokey/issues/58 IDK if it will help.

setup.py no longer refers to python.dbus, so that proposed change was made.

See setup.py lines 36 - 38, 191. It's testing for Python 3.5 or greater and probably needs to be bumped, but it won't help this issue. Looks like you got these.

autokey/debian/py3dist-overrides contains:

dbus-python python3-dbus
python-xlib python3-xlib

/autokey/.github/workflows/python-test.yml 
158        pip install dbus-python gobject pygobject PyQt5 qscintilla
Looks like you got this.

which contains both offenders, but IDK what that file is used for. Looks important.

*** Did you catch this one? ***

autokey/pip-requirements.txt contains:

dbus-python
pyinotify
python-xlib
PyGObject
PyQt5
QScintilla
pyhamcrest

again, both offenders. Looks important.

Looks like gobject and PyGObject don't appear anywhere else.

**Comment by Elliria** [comment]
Thank you for all the research. I didn't catch that one and I'll see what I can figure out from all of the things you found.

By the way, you'll want to make sure you're looking at [AutoKey's develop branch](https://github.com/autokey/autokey/tree/develop) so you can examine the most current files and the ones I'm trying to change.

**Comment by Elliria** [comment]
The #58 issue looks interesting. Two of the packages most often recommended there (`libdbus-1-dev` and `libdbus-glib-1-dev`) are already AutoKey dependencies, but a comment by [ding3my](https://github.com/autokey/autokey/issues/58#issuecomment-318249538) recommended installing `dbus-python-devel` and that isn't in the list of AutoKey's dependencies.

I did a search for it in the Ubuntu repositories and it doesn't exist, but my package manager is offering me `python-dbus-dev` when I did that search. Adding that package to the **apt-requirements.txt** file and the **pip-requirements.txt** file without making any other changes to my current fork (which only has its Python versions updated) doesn't solve the issue, but something else surely has to happen.

**Comment by josephj11** [comment]
Yeah. I realized that after I did it and saw that you had already fixed most of them. I'll have to clone it and figure out how to update it periodically.

The trick (that you already see) is that adding things has to be done on GitHub as well as locally because the tests run there and not on your local machine.

python-dbus-dev is current (bullseye) and future (bookworm). It talks about "compiling" and I'm not sure how that works in Python although I have run into `.pyc` files before. I suspect that a lot of the dbus stuff is actually C code that is linked into Python, but I don't know what to do with that.

**Comment by Elliria** [comment]
You can either download the zip (making sure you're on the **develop** branch before clicking the **Code** button to grab the zip) or by cloning the develop branch. The clone command I had been using didn't work. It clones the **master** branch. I had to do a bit of research to figure out how to insist on the **develop** branch.

To clone AutoKey's **develop** branch, you'll want to use:
```
git clone --branch develop --single-branch https://github.com/autokey/autokey.git
```

To clone AutoKey's **master** branch, you'll want to use:
```
git clone https://github.com/autokey/autokey.git
```

I added a note about this to section 6 of my Gist page for quick reference.

**Comment by Elliria** [comment]
Oh, and I was wrong about the tests passing locally and failing on GitHub. Once I cloned the **develop** version and ran the tests, they failed for me locally as well.

**Comment by Elliria** [comment]
I wasn't familiar with **.pyc** files, either. According to **geeksforgeeks.org**, **.pyc** files contain 'the “compiled bytecode” of the imported module/program so that the “translation” from source code to bytecode can be skipped on subsequent imports of the *. py file. Having a *. pyc file saves the compilation time of converting the python source code to byte code, every time the file is imported.'

**Comment by Elliria** [comment]
This is what I had originally thought should be done, but it may not be right, so I moved it out of the bug report above and am putting it into this comment instead:


Check the [GitHub versions manifest](https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json) for all versions marked with the current Ubuntu release number as their **platform_version** and, based on the currently-supported versions, make these changes:

1. [Optional] In the **setup.cfg** file:
   - [Line 23](https://github.com/autokey/autokey/blob/develop/setup.cfg#L23): change **py36,py39** to **py37,py310**
2. In the **setup.py** file:
   - [Line 36](https://github.com/autokey/autokey/blob/develop/setup.py#L36): change **3, 5, 0** to **3, 7, 15**
   - [Line 37](https://github.com/autokey/autokey/blob/develop/setup.py#L37): change **3.5** to **3.7.15**
   - [Line 140](https://github.com/autokey/autokey/blob/develop/setup.py#L140): change **3.5** to **3.7.15**
   - [Line 212](https://github.com/autokey/autokey/blob/develop/setup.py#L212): change **3.5** to **3.7**
3. In the **/.github/workflows/build.yml** file:
   - [Line 19](https://github.com/autokey/autokey/blob/develop/.github/workflows/build.yml#L19): change **3.9** to **3.10**
   - [Line 67](https://github.com/autokey/autokey/blob/develop/.github/workflows/build.yml#L67): change **3.9** to **3.10**
   - [Line 113](https://github.com/autokey/autokey/blob/develop/.github/workflows/build.yml#L113): change **3.9** to **3.10**
   - [Line 128](https://github.com/autokey/autokey/blob/develop/.github/workflows/build.yml#L128): change **3.9** to **3.10**
4. In the **/.github/workflows/python-test.yml** file:
   - [Line 24](https://github.com/autokey/autokey/blob/develop/.github/workflows/python-test.yml#L22): change **3.6, 3.7, 3.9** to **"3.7", "3.10"**
   - [Line 86](https://github.com/autokey/autokey/blob/develop/.github/workflows/python-test.yml#L84): change **3.6, 3.7, 3.9** to **"3.7", "3.10"**
   - [Line 145](https://github.com/autokey/autokey/blob/develop/.github/workflows/python-test.yml#L144): change **3.6, 3.7, 3.9** to **"3.7", "3.10"**
5. In the **/debian/rules** file:
   - [Line 4](https://github.com/autokey/autokey/blob/develop/debian/rules#L4): change **3.8** to **3.10.1**

**Comment by Elliria** [comment]
If pull request #786 gets merged in, then everything except for step 5 is done.

**Comment by Elliria** [comment]
I reopened this since step 5 hasn't been completed yet. It will have to wait until I contact someone on the Debian team to find out what the change ought to be.
